Tests/registrationTests.js

```diff
@@ -0,0 +1,137 @@
+var homePage = require('../Pages/homePage.js'),
+    itemPage = require ("../Pages/itemPage"),
+    registerPage = require ("../Pages/registerPage"),
+    data = require("../Data/data.js");
+browser.waitForAngularEnabled(false);
+browser.ignoreSynchronization = true;
+browser.manage().window().maximize();
+
+var homePage = require('../Pages/homePage.js'),
+    itemPage = require ("../Pages/itemPage"),
+    registerPage = require ("../Pages/registerPage"),
+    data = require("../Data/data.js");
+browser.waitForAngularEnabled(false);
+browser.ignoreSynchronization = true;
+browser.manage().window().maximize();
+
+describe("002: Registration", function(){
+    beforeEach(() => {
+        homePage.openPageURL(data.homepageLink)
+        .then(() => homePage.clickOnCreateAnAccount())
+    })
+
+    it ("001: User is able to create account with valid credentials", function(){
+        registerPage.enterFirstName(data.firstNameTester)
+            .then(() => registerPage.enterLastName(data.lastNameTester))
+            .then(() => registerPage.enterEmail())
+            .then(() => registerPage.enterPassword(data.passwordTester))
+            .then(() => registerPage.enterConfirmPassword(data.passwordTester))
+            .then(() => registerPage.clickOnRegisterButton())
+            .then(() => itemPage.waitForSuccessMessage())
+            .then(() => registerPage.getRegistrationMessage())
+            .then((registrationMessage) => registerPage.validateRegistrationMessage(registrationMessage))
+    }) 
+
+    it ("002: User isn't able to create account with invalid credentials", function (){
+        registerPage.enterFirstName(data.fiveSpaces)
+            .then(() => registerPage.enterLastName(data.fiveSpaces))
+            .then(() => registerPage.enterEmail())
+            .then(() => registerPage.enterPassword(data.fiveSpaces))
+            .then(() => registerPage.enterConfirmPassword(data.fiveSpaces))
+            .then(() => registerPage.clickOnRegisterButton())
+            .then(() => registerPage.getFirstNameValidationMessage())
```

Comment:
Try to put connected logic to one method and move that method to page object or to separated class "UITasks".
For example, here "   .then(() => registerPage.getFirstNameValidationMessage())
            .then((invalidFirstName) => registerPage.validateRequiredFirstName(invalidFirstName))
            .then(() => registerPage.getLastNameValidationMessage())
            .then((invalidLastName) => registerPage.validateRequiredLastName(invalidLastName))
            .then(() => registerPage.getPasswordValidationMessage())
            .then((invalidPassword) => registerPage.validateInvalidPassword(invalidPassword))"
can be one method something like "validateMessagesRegistration"

Reply:
I added this logic at places where it could be done, so now tests contain much less code for execution of the methods